csharp/extractor/Semmle.Extraction/Extractor/BinaryLogExtractionContext.cs

```diff
@@ -0,0 +1,60 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Semmle.Util.Logging;
+
+namespace Semmle.Extraction
+{
+    public class BinaryLogExtractionContext : ExtractionContext
+    {
+        private readonly IEnumerable<SyntaxTree> generatedSyntaxTrees;
+        private readonly string compilationIdentifier;
+        private readonly string generatedFolderName;
+
+        public BinaryLogExtractionContext(string cwd, string[] args, string outputPath,
+            IEnumerable<SyntaxTree> generatedSyntaxTrees, string compilationIdentifier,
+            ILogger logger, PathTransformer pathTransformer, bool isQlTest)
+            : base(cwd, args, outputPath, [], logger, pathTransformer, ExtractorMode.BinaryLog, isQlTest)
+        {
+            this.generatedSyntaxTrees = generatedSyntaxTrees;
+            this.compilationIdentifier = compilationIdentifier;
+
+            // Compute a unique folder name for the generated files:
+            generatedFolderName = "generated";
+
+            if (Directory.Exists(generatedFolderName))
```

Comment:
This code will check for `generated` in the current working directory. Did you mean to do something like `Path.Combine(cwd, generatedFolderName)`?

Reply:
Yes, this is on purpose. The current working directory will point to the `--source-root` specified in the codeql database creation command. In practice this will be the repo root that you're analysing. All the processed source files are stored in the source archive in the codeql database. And this logic is making sure that we're assigning the generated ASTs paths that don't collide with any already existing source files.

The integration test results in `csharp/ql/integration-tests/all-platforms/binlog/Files.expected` show some sample paths, for example:
```
generated/b/test.csproj (net8.0)/System.Text.RegularExpressions.Generator/System.Text.RegularExpressions.Generator.RegexGenerator/RegexGenerator.g.cs
```

We could also use `Path.Combine(cwd, generatedFolderName)`, then the paths would point to locations closer to their original `.csproj` files, such as 
```
b/generated/test.csproj (net8.0)/System.Text.RegularExpressions.Generator/System.Text.RegularExpressions.Generator.RegexGenerator/RegexGenerator.g.cs
```

I think the root level `generated` folder should work, and then all these generated ASTs are assigned to a single folder.